src/core/system_parameters/beidou_cnav1_navigation_message.cc

```diff
@@ -0,0 +1,605 @@
+            cnav1_almanac[i_alm_satellite_PRN - 1].Omega_dot = static_cast<double>(read_navigation_signed(subframe3, Omega_dot_4)) * BEIDOU_cnav1_PI * TWO_N33;  //[pi/s]
+            cnav1_almanac[i_alm_satellite_PRN - 1].omega = static_cast<double>(read_navigation_signed(subframe3, omega_4)) * BEIDOU_cnav1_PI * TWO_N15;          //[pi]
+            cnav1_almanac[i_alm_satellite_PRN - 1].M_0 = static_cast<double>(read_navigation_signed(subframe3, M_0_4)) * BEIDOU_cnav1_PI * TWO_N15;              //[pi]
+            cnav1_almanac[i_alm_satellite_PRN - 1].a_f0 = static_cast<double>(read_navigation_signed(subframe3, a_f0_4)) * TWO_N20;                              //[s]
+            cnav1_almanac[i_alm_satellite_PRN - 1].a_f1 = static_cast<double>(read_navigation_signed(subframe3, a_f1_4)) * TWO_N37;                              //[s/s]
+            cnav1_almanac[i_alm_satellite_PRN - 1].Health = static_cast<double>(read_navigation_unsigned(subframe3, Health_4));                                  //[dimensionless] 8th(MSB):Satellite, 7th:B1C, 6th:B2a, 5th~1st:reserve, 0:normal/health, 1:abnormal
+            // Midi Almanac Parameters End
+            
+            cnav1_ephemeris.Rev = static_cast<double>(read_navigation_unsigned(subframe3, Rev_4));
+            
+            flag_TOW_set = true;
+            flag_TOW_4 = true;
+            flag_ephemeris_mes_type_4 = true;
+            flag_almanac_mes_type_4 = true;
+            
+            break;
+
+
+        default:
+            LOG(INFO) << "BEIDOU CNAV1: Invalid String ID of received. Received " << i_frame_mes_type
+                      << ", but acceptable values are 1, 2, 3 and 4";
+
+            break;
+        }  // switch string ID
+
+    return i_frame_mes_type;
+}
+
+Beidou_Cnav1_Ephemeris Beidou_Cnav1_Navigation_Message::get_ephemeris()
+{
+    return cnav1_ephemeris;
+}
+
+Beidou_Cnav1_Utc_Model Beidou_Cnav1_Navigation_Message::get_utc_model()
+{
+    return cnav1_utc_model;
+}
+
+Beidou_Cnav1_Iono Beidou_Cnav1_Navigation_Message::get_iono()
+{
+    return cnav1_iono;
+}
+
+Beidou_Cnav1_Almanac Beidou_Cnav1_Navigation_Message::get_almanac(uint32_t satellite_slot_number)
+{
+    return cnav1_almanac[satellite_slot_number - 1];
+}
+
+bool Beidou_Cnav1_Navigation_Message::have_new_ephemeris()  //Check if we have a new ephemeris stored in the galileo navigation class
+{
+    bool new_eph = false;
+    // We need to make sure we have received the ephemeris info plus the time info
+    if (flag_ephemeris_mes_SF2 == true)
+        {
+            if (d_previous_tb != cnav1_ephemeris.IODE)
+                {
+                    flag_ephemeris_mes_SF2 = false;  // clear the flag
+                    flag_all_ephemeris = true;
+                    // Update the time of ephemeris information
+                    d_previous_tb = cnav1_ephemeris.IODE;
+                    DLOG(INFO) << "Beidou Cnav1 Ephemeris from Subframe 2 have been received and belong to the same batch" << std::endl;
```

Comment:
for printing make it BeiDou rather than Beidou

Reply:
Made appropriate changes.